examples/all-clusters-app/esp32/main/include/ShellCommands.h

```diff
@@ -0,0 +1,209 @@
+    {
+        if (argc != 1)
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        bool value;
+        chip::app::Clusters::OnOff::Attributes::OnOff::Get(atoi(argv[0]), &value);
+        chip::app::Clusters::OnOff::Attributes::OnOff::Set(atoi(argv[0]), !value);
+        return CHIP_NO_ERROR;
+    }
+
+    static Shell::Engine sSubShell;
+};
+
+class CASECommands
+{
+public:
+    // delete the copy constructor
+    CASECommands(const CASECommands &) = delete;
+    // delete the move constructor
+    CASECommands(CASECommands &&) = delete;
+    // delete the assignment operator
+    CASECommands & operator=(const CASECommands &) = delete;
+
+    static CASECommands & GetInstance()
+    {
+        static CASECommands instance;
+        return instance;
+    }
+
+    // Register the CASESession commands
+    void Register();
+
+    void SetFabricInfo(FabricInfo * fabricInfo) { mFabricInfo = fabricInfo; }
+    void SetNodeId(NodeId nodeId) { mNodeId = nodeId; }
+    void SetOnConnecting(bool onConnecting) { mOnConnecting = onConnecting; }
+    FabricInfo * GetFabricInfo(void) { return mFabricInfo; }
+    NodeId GetNodeId(void) { return mNodeId; }
+    bool GetOnConnecting(void) { return mOnConnecting; }
+private:
+    CASECommands() {}
+    static void OnConnected(void * context, OperationalDeviceProxy * deviceProxy)
+    {
+        streamer_printf(streamer_get(), "Establish CASESession Success!\r\n");
+        GetInstance().SetOnConnecting(false);
+    }
+
+    static void OnConnectionFailure(void * context, PeerId peerId, CHIP_ERROR error)
+    {
+        streamer_printf(streamer_get(), "Establish CASESession Failure!\r\n");
+        GetInstance().SetOnConnecting(false);
+    }
+
+    static void ConnectToNode(intptr_t arg)
+    {
+        CASECommands * caseCommand              = reinterpret_cast<CASECommands *>(arg);
+        Server * server                         = &(chip::Server::GetInstance());
+        CASESessionManager * caseSessionManager = server->GetCASESessionManager();
+        if (caseSessionManager == nullptr)
+        {
+            ChipLogError(SecureChannel, "Can't get the CASESessionManager");
+            return;
+        }
+        caseSessionManager->FindOrEstablishSession(caseCommand->GetFabricInfo()->GetPeerIdForNode(caseCommand->GetNodeId()),
+                                                   &sOnConnectedCallback, &sOnConnectionFailureCallback);
+    }
+
+    static CHIP_ERROR ConnectToNodeHandler(int argc, char ** argv)
+    {
+        if (GetInstance().GetOnConnecting())
+        {
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+        const FabricIndex fabricIndex = static_cast<FabricIndex>(strtoul(argv[0], nullptr, 10));
+        FabricInfo * fabricInfo       = Server::GetInstance().GetFabricTable().FindFabricWithIndex(fabricIndex);
+
+        if (fabricInfo == nullptr)
+        {
+            streamer_printf(streamer_get(), "Can't get fabric information from FabricIndex\r\n");
```

Comment:
I have found that FindFabricWithIndex can return uncommissioned fabrics, so long as index is < CHIP_CONFIG_MAX_DEVICE_ADMINS.

It does not appear this will be a problem in this PR.  I expect if this happened, you would end up with a peer ID on no-fabric-index-0, and that CASESessionManager would understand this and error out appropriately.

So this is FYI.

The behavior I saw may also have been chip-tool specific.